### PR TITLE
feat: SSR support

### PR DESCRIPTION
**Pull request by tomas-c**
Before these changes, the Frame and Element components delayed rendering user components until after the first render due to their reliance on useEffect. This meant that (a) server-side rendering did not work (b) pages with many deeply nested components took many render iterations to fully render a page which on slower devices would cause a visible layout shift.

After these changes neither Element nor Frame rely on useEffect.

Resolves:
- #140
- #200 
- #42

### CODE REVIEW

**Comment by hugominas** [comment]
@tomas-c great work, this is such a needed feature @prevwong would be great to have your eyes on this PR

File: packages/core/src/render/Frame.tsx [hunk 1]
```diff
@@ -44,21 +44,21 @@ export const Frame: React.FC<React.PropsWithChildren<FrameProps>> = ({
...
+    actions.history.ignore().deserialize(initialState.current.initialData);
+    isInitialLoadedRef.current = true;
+  }
 
-    if (initialData) {
-      actions.history.ignore().deserialize(initialData);
-      return;
-    }
+  useEffect(() => {
+    const { initialChildren } = initialState.current;
```
Comment (prevwong) [inline packages/core/src/render/Frame.tsx:55]: I think you we can simplify this to:
```tsx
  const isLoaded = useRef(false);

  if (!isLoaded.current) {
    const initialData = data || json;

    if (initialData) {
      actions.history.ignore().deserialize(initialData);
    } else {
      const rootNode = React.Children.only(children) as React.ReactElement;

      const node = query.parseReactElement(rootNode).toNodeTree((node, jsx) => {
        if (jsx === rootNode) {
          node.id = ROOT_NODE;
        }
        return node;
      });

      actions.history.ignore().addNodeTree(node);
    }
```

so we can remove the `useEffect` hook completely

**Review by prevwong — COMMENTED** [review]
Looks good, just left a suggestion

File: packages/core/src/render/Frame.tsx (hunk 1 above)
Comment (tomas-c) [reply packages/core/src/render/Frame.tsx]: Thanks! I've applied the changes. This was where I wasn't sure if Creaft.js somehow relied on the children parsing being done after rendering. As far as I can tell looking in more detail - it doesn't. And tests do pass and I couldn't see anything breaking while playing around with the examples in the repo and my own app. So hopefully good to go!

Comment (prevwong) [reply packages/core/src/render/Frame.tsx]: Yeap, Craft just parses the children elements if there was no `data` value present - so in this way, the children elements are used to create the default/initial content for the editor

**Review by prevwong — APPROVED** [review]
Thank you! 💯

**Pull request merged**